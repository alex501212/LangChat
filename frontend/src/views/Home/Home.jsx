import React, { useEffect } from "react";
import "./Home.scss";
import { Button, ButtonGroup, Flex, Heading } from "@chakra-ui/react";
import { useNavigate } from "react-router-dom";
import bg from "../../bg.png";

const Home = () => {
  const navigate = useNavigate();

  useEffect(() => {
    sessionStorage.removeItem("token");
  }, []);

  return (
    <div>
      <Flex
        boxShadow="inner"
        w="100vw"
        h="90vh"
        justifyContent="center"
        alignItems="center"
        style={{
          background: `url(${bg})`,
          backgroundSize: "1920px 940px",
          backgroundPosition: "0px -94px",
          backgroundColor: "#718096",
        }}
        p="7"
      >
        <ButtonGroup variant="outline" spacing="20" mb="10">
          <Heading size="2xl" textColor={"white"}>
            Welcome to LangChat
          </Heading>
          <Button
            right={"40px"}
            boxShadow="xl"
            colorScheme="blue"
            variant="solid"
            size="lg"
            onClick={() => navigate("/login")}
          >
            Login
          </Button>
          <Button
            right={"40px"}
            boxShadow="xl"
            colorScheme="blue"
            variant="solid"
            size="lg"
            onClick={() => navigate("/register")}
          >
            Register
          </Button>
        </ButtonGroup>
      </Flex>
    </div>
  );
};

export default Home;
