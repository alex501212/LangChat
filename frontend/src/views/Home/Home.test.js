import { cleanup, render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import Home from "./Home";

afterEach(() => {
  cleanup();
});

it("matches snapshot", () => {
  const container = renderer
    .create(
      <BrowserRouter>
        <Home />
      </BrowserRouter>
    )
    .toJSON();
  expect(container).toMatchSnapshot();
});

it("renders all text", () => {
  const view = render(
    <BrowserRouter>
      <Home />
    </BrowserRouter>
  );
  expect(screen.getByText("Welcome to LangChat")).toBeInTheDocument();
  expect(screen.getByText("Login")).toBeInTheDocument();
  expect(screen.getByText("Register")).toBeInTheDocument();
});
