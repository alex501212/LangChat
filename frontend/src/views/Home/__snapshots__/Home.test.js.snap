// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    className="css-1b4xnu2"
  >
    <div
      className="chakra-button__group css-goy9dw"
      data-orientation="horizontal"
      role="group"
    >
      <button
        className="chakra-button css-1fysbuj"
        onClick={[Function]}
        type="button"
      >
        Login
      </button>
      <button
        className="chakra-button css-1fysbuj"
        onClick={[Function]}
        type="button"
      >
        Register
      </button>
    </div>
  </div>
</div>
`;
